core/lib/config/src/configs/contracts/chain.rs

```diff
@@ -1,144 +1,394 @@
+            CONTRACTS_L2_LEGACY_SHARED_BRIDGE_ADDR="0x8656770FA78c830456B00B4fFCeE6b1De0e1b888"
+            CONTRACTS_BRIDGEHUB_PROXY_ADDR="0x35ea7f92f4c5f433efe15284e99c040110cf6297"
+            CONTRACTS_STATE_TRANSITION_PROXY_ADDR="0xd90f1c081c6117241624e97cb6147257c3cb2097"
+            CONTRACTS_TRANSPARENT_PROXY_ADMIN_ADDR="0xdd6fa5c14e7550b4caf2aa2818d24c69cbc347e5"
+            CONTRACTS_BASE_TOKEN_ADDR="0x0000000000000000000000000000000000000001"
+            CONTRACTS_CHAIN_ADMIN_ADDR="0xdd6fa5c14e7550b4caf2aa2818d24c69cbc347ff"
+            CONTRACTS_L2_DA_VALIDATOR_ADDR="0xed6fa5c14e7550b4caf2aa2818d24c69cbc347ff"
+            CONTRACTS_L2_TIMESTAMP_ASSERTER_ADDR="0x0000000000000000000000000000000000000002"
+
+            CONTRACTS_L1_BYTECODES_SUPPLIER_ADDR="F00B988a98Ca742e7958DeF9F7823b5908715f4a"
+            CONTRACTS_L1_WRAPPED_BASE_TOKEN_STORE="0x35ea7f92f4c5f433efe15284e99c040110cf6297"
+            CONTRACTS_SERVER_NOTIFIER_ADDR="F00B988a98Ca742e7958DeF9F7823b5908715f4a"
+            CONTRACTS_BASE_TOKEN_ASSET_ID="0x00000000000000000000000000000000000000000000000000000000075bcd15"
+            CONTRACTS_NO_DA_VALIDIUM_L1_VALIDATOR_ADDR="0x34782eE00206EAB6478F2692caa800e4A581687b"
+            CONTRACTS_L2_MULTICALL3_ADDR="0x0000000000000000000000000000000000010002"
+
+            CONTRACTS_BRIDGES_SHARED_L1_ADDRESS="0x8656770FA78c830456B00B4fFCeE6b1De0e1b888"
+            CONTRACTS_BRIDGES_SHARED_L2_ADDRESS="0x8656770FA78c830456B00B4fFCeE6b1De0e1b888"
+            CONTRACTS_BRIDGES_ERC20_L1_ADDRESS="0x8656770FA78c830456B00B4fFCeE6b1De0e1b888"
+            CONTRACTS_BRIDGES_ERC20_L2_ADDRESS="0x8656770FA78c830456B00B4fFCeE6b1De0e1b888"
+            CONTRACTS_BRIDGES_WETH_L1_ADDRESS="0x8656770FA78c830456B00B4fFCeE6b1De0e1b888"
+            CONTRACTS_BRIDGES_WETH_L2_ADDRESS="0x8656770FA78c830456B00B4fFCeE6b1De0e1b888"
+        "#;
+        let env = Environment::from_dotenv("test.env", env).unwrap();
+
+        let schema = create_schema();
+        let repo = ConfigRepository::new(&schema).with(env);
+        let config: ContractsConfig = repo.single().unwrap().parse().unwrap();
+        assert_eq!(config, expected_config());
+    }
+
+    #[test]
+    fn parsing_from_yaml() {
+        let yaml = r#"
+          contracts:
+            ecosystem_contracts:
+              bridgehub_proxy_addr: 0x35ea7f92f4c5f433efe15284e99c040110cf6297
+              state_transition_proxy_addr: 0xd90f1c081c6117241624e97cb6147257c3cb2097
+              transparent_proxy_admin_addr: 0xdd6fa5c14e7550b4caf2aa2818d24c69cbc347e5
+              l1_bytecodes_supplier_addr: F00B988a98Ca742e7958DeF9F7823b5908715f4a
+              l1_wrapped_base_token_store: 0x35ea7f92f4c5f433efe15284e99c040110cf6297
+              server_notifier_addr: F00B988a98Ca742e7958DeF9F7823b5908715f4a
+              # NOT PARSED: validator_timelock_addr: 0xF00B988a98Ca742e7958DeF9F7823b5908715f4a
```

Comment:
wdym? why?

Reply:
We have `validator_timelock_addr` in two places in our config files, in `l1` and in `ecosystem`. Value in `l1` is parsed and used, the other is not. It's super confusing and it has led to misconfiguration a few times already. Conceptually, it should be a part of `ecosystem` contracts. But I would prefer not to touch it in this PR and tackle it separately

Reply:
Oh, yeah, i remember it. 
it was an artifact one day. 
probably right after this pr we could tackle such missconfigs